lib/datasource/remote/model/token_model.dart

```diff
@@ -1,33 +1,109 @@
 import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:json_schema2/json_schema2.dart';
+import 'package:seeds/datasource/remote/api/tokenmodels_repository.dart';
+import 'package:seeds/datasource/remote/firebase/firebase_remote_config.dart';
+import 'package:seeds/domain-shared/shared_use_cases/get_token_models_use_case.dart';
+import 'package:seeds/screens/wallet/components/tokens_cards/components/currency_info_card.dart';
 
-class TokenModel extends Equatable {
-  static const allTokens = [seedsToken, husdToken, hyphaToken, localScaleToken, starsToken, telosToken];
 
+class TokenModel extends Equatable {
+  static const seedsEcosysUsecase = 'seedsecosys';
+  static List<TokenModel> allTokens = [seedsToken];
+  static JsonSchema? tmastrSchema;
   final String chainName;
   final String contract;
   final String symbol;
   final String name;
-  final String backgroundImage;
-  final String logo;
+  final String backgroundImageUrl;
+  final String logoUrl;
   final String balanceSubTitle;
   final int precision;
+  final List<String>? usecases;
 
   String get id => "$contract#$symbol";
 
+  ImageProvider get backgroundImage {
+    return
+      backgroundImageUrl.startsWith("assets") ?
+      AssetImage(backgroundImageUrl) as ImageProvider :
+      NetworkImage(backgroundImageUrl);
+  }
+  ImageProvider get logo {
+    return
+      logoUrl.startsWith("assets") ?
+      AssetImage(logoUrl) as ImageProvider :
+      NetworkImage(logoUrl);
+  }
+
   const TokenModel({
     required this.chainName,
     required this.contract,
     required this.symbol,
     required this.name,
-    required this.backgroundImage,
-    required this.logo,
+    required this.backgroundImageUrl,
+    required this.logoUrl,
     required this.balanceSubTitle,
     this.precision = 4,
+    this.usecases,
   });
 
-  factory TokenModel.fromSymbol(String symbol) {
-    return allTokens.firstWhere((e) => e.symbol == symbol);
+  static Future<Result<void>> installSchema() async {
+    final result = await TokenModelsRepository().getSchema();
+      if(result.isValue) {
+        final tmastrSchemaMap = result.asValue!.value;
+        tmastrSchema = JsonSchema.createSchema(tmastrSchemaMap);
+        return Result.value(null);
+      }
+      print('Error getting Token Master schema from chain');
+      return result;
+  }
+
+  static TokenModel? fromJson(Map<String,dynamic> data) {
+    final Map<String,dynamic> parsedJson = json.decode(data["json"]);
+    bool extendJson(String dataField, String jsonField) {
+      final jsonData = parsedJson[jsonField];
+      if( jsonData != null && jsonData != data[dataField] ) {
+        print('${data[dataField]}: mismatched $dataField in json'
+            ' $jsonData, ${data[dataField]}');
+        return false;
+      } else {
+        parsedJson[jsonField] ??= data[dataField];
+        return true;
+      }
+    }
+    if (!( extendJson("chainName", "chain") &&
```

Comment:
I don't understand the use of this mapping from json to data field?

Why not name them the same? 

Am I missing something?

Reply:
There's an obscure reason for this, though it might not be an especially good one. The more-or-less "industry standard" for eosio token metadata was affiliated with the bloks.io explorer https://github.com/eoscafe/eos-airdrops . For example the SEEDS record there is
```
    {
        "name": "Seeds",
        "logo": "https://raw.githubusercontent.com/eoscafe/eos-airdrops/master/logos/seeds.png",
        "logo_lg": "https://raw.githubusercontent.com/eoscafe/eos-airdrops/master/logos/seeds-lg.png",
        "symbol": "SEEDS",
        "account": "token.seeds",
        "chain": "telos"
    }
```
Originally I was trying to maintain some compatibility, as Rainbow token creators probably would like their token logos to appear in block explorers as well.

Reply:
Leaving this as is